### PR TITLE
Added carthage support

### PR DESCRIPTION
**Pull request by delannoyk**
Hi Javi,

I just added a framework target (for which scheme's is shared), added JSDecoupledAppDelegate.m in target's source and made JSDecoupledAppDelegate.h as public header.

### CODE REVIEW

File: JSDecoupledAppDelegate_SampleApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -289,6 +358,62 @@
...
+				DYLIB_CURRENT_VERSION = 1;
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				ENABLE_TESTABILITY = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
+				INFOPLIST_FILE = JSDecoupledAppDelegate/Info.plist;
+				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
+				IPHONEOS_DEPLOYMENT_TARGET = 9.2;
```
Comment (JaviSoto) [inline JSDecoupledAppDelegate_SampleApp.xcodeproj/project.pbxproj]: We could lower this to 8.0, right?

File: JSDecoupledAppDelegate_SampleApp.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -289,6 +358,62 @@
...
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				MTL_ENABLE_DEBUG_INFO = NO;
+				PRODUCT_BUNDLE_IDENTIFIER = com.javisoto.JSDecoupledAppDelegate;
+				PRODUCT_NAME = "$(TARGET_NAME)";
```
Comment (JaviSoto) [inline JSDecoupledAppDelegate_SampleApp.xcodeproj/project.pbxproj]: What's the name of the target? We may want this to be "JSDecoupledAppDelegate" instead.

**Comment by JaviSoto** [comment]
Just a couple comments :)
Did you try this with Carthage yet?

Thanks a lot for this!! :)

File: JSDecoupledAppDelegate_SampleApp.xcodeproj/project.pbxproj (hunk 1 above)
Comment (delannoyk) [inline JSDecoupledAppDelegate_SampleApp.xcodeproj/project.pbxproj]: Yes!

File: JSDecoupledAppDelegate_SampleApp.xcodeproj/project.pbxproj (hunk 2 above)
Comment (delannoyk) [inline JSDecoupledAppDelegate_SampleApp.xcodeproj/project.pbxproj]: Target's name is already "JSDecoupledAppDelegate".

**Comment by delannoyk** [comment]
Corrected deployment version :+1: 
Also, yes, I'm currently using my fork as source but everything works just fine.

**Comment by delannoyk** [comment]
Hi Javi,
Is there anything left for me to do for this PR? :)

**Comment by JaviSoto** [comment]
I'm so sorry it took so long!!! It got lost in my inbox.

This is awesome, thank you so much for contributing! <3

**Pull request merged**